nighthawk/source/client/stream_decoder.cc

```diff
@@ -0,0 +1,55 @@
+#include "nighthawk/source/client/stream_decoder.h"
+
+#include "common/http/http1/codec_impl.h"
+#include "common/http/utility.h"
+
+namespace Nighthawk {
+namespace Client {
+
+void StreamDecoder::decodeHeaders(Envoy::Http::HeaderMapPtr&& headers, bool end_stream) {
+  ASSERT(!complete_);
+  complete_ = end_stream;
+  response_headers_ = std::move(headers);
+  if (complete_) {
+    onComplete(true);
+  }
+}
+
+void StreamDecoder::decodeData(Envoy::Buffer::Instance&, bool end_stream) {
+  ASSERT(!complete_);
+  complete_ = end_stream;
+  if (complete_) {
+    onComplete(true);
+  }
+}
+
+void StreamDecoder::decodeTrailers(Envoy::Http::HeaderMapPtr&&) { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
```

Comment:
How come we never see trailers?

Reply:
Fixed, plus I added some standalone testing for StreamDecoder in this PR.